katcp/resource_client.py

```diff
@@ -881,33 +881,37 @@ def list_sensors(self, filter="", strategy=False, status="",
     def set_sensor_strategy(self, sensor_name, strategy_and_parms):
         sensor_name = resource.escape_name(sensor_name)
         sensor_obj = getattr(self.sensor, sensor_name, None)
-        for child_name in dict.keys(self.children):
+        if sensor_obj:
+            # The sensor exists, so set the strategy.
+            child_name = sensor_obj.parent_name
```

Comment:
I agree with @ludwigschwardt that we should distinguish who's-child are we manipulating here.